### ISSUE TITLE
Error in over(x = as(object = x, Class = "SpatialPolygons"), y = as(object = y,  :    use sf or terra functions

### ISSUE THREAD

**Issue by cgoneill** [satijalab/seurat#7872]
When I run `Crop()`, even an exact copy of code I have previously run without issue, instead of a Large FOV object, I get the following error message: 

```r
> wt1_cropped_coords1 <- Crop(Dlx3_obj[["fov"]], 
+                             x = c(1550, 2050), 
+                             y = c(3900, 4200), 
+                             coords = "tissue")
Error in over(x = as(object = x, Class = "SpatialPolygons"), y = as(object = y,  : 
  use sf or terra functions
In addition: Warning messages:
1: In over(x = as(object = x, Class = "SpatialPolygons"), y = as(object = y,  :
  No rgeos support in sp from October 2023;
see https://r-spatial.org/r/2023/05/15/evolution4.html
2: 'over' is deprecated.
Use 'sf::st_intersects or sf::aggregate' instead.
See help("Deprecated") 
```

When I run the following code from the Xenium vignette with the example dataset, I get the same error message: 

```r
> cropped.coords <- Crop(xenium.obj[["fov"]], 
+                        x = c(1200, 2900), 
+                        y = c(3750, 4550), 
+                        coords = "plot")
Error in over(x = as(object = x, Class = "SpatialPolygons"), y = as(object = y,  : 
  use sf or terra functions
In addition: Warning messages:
1: In over(x = as(object = x, Class = "SpatialPolygons"), y = as(object = y,  :
  No rgeos support in sp from October 2023;
see https://r-spatial.org/r/2023/05/15/evolution4.html
2: 'over' is deprecated.
Use 'sf::st_intersects or sf::aggregate' instead.
See help("Deprecated") 
```

Here is the `sessionInfo()` output - I'm using RStudio on Rocky8/RHEL8. 

```r
> sessionInfo()
R version 4.3.0 (2023-04-21)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Rocky Linux 8.7 (Green Obsidian)

Matrix products: default
BLAS/LAPACK: /usr/local/intel/2022.1.2.146/mkl/2022.0.2/lib/intel64/libmkl_rt.so.2;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: America/New_York
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] ggpubr_0.6.0                   foreach_1.5.2                  jackstraw_1.3.8                BiocParallel_1.34.2            BiocSingular_1.16.0           
 [6] biomaRt_2.56.1                 scran_1.28.2                   bluster_1.10.0                 scater_1.28.0                  scuttle_1.10.2                
[11] nngeo_0.4.7                    sf_1.0-12                      vroom_1.6.4                    arrow_12.0.1                   DropletUtils_1.20.0           
[16] SpatialFeatureExperiment_1.2.3 SpatialExperiment_1.10.0       SingleCellExperiment_1.22.0    SummarizedExperiment_1.30.2    Biobase_2.60.0                
[21] GenomicRanges_1.52.0           GenomeInfoDb_1.36.4            IRanges_2.34.1                 S4Vectors_0.38.2               BiocGenerics_0.46.0           
[26] MatrixGenerics_1.12.3          matrixStats_1.0.0              paletteer_1.5.0                glmGamPoi_1.12.2               openxlsx_4.2.5.2              
[31] SeuratWrappers_0.3.19          Matrix_1.6-0                   clustree_0.5.0                 ggraph_2.1.0                   Voyager_1.2.7                 
[36] spacexr_2.2.1                  cowplot_1.1.1                  magrittr_2.0.3                 lubridate_1.9.2                forcats_1.0.0                 
[41] stringr_1.5.0                  dplyr_1.1.3                    purrr_1.0.2                    readr_2.1.4                    tidyr_1.3.0                   
[46] tibble_3.2.1                   ggplot2_3.4.3                  tidyverse_2.0.0                parallelly_1.36.0              future_1.33.0                 
[51] Seurat_4.9.9.9067              SeuratObject_4.9.9.9091        sp_2.1-0                      

loaded via a namespace (and not attached):
  [1] R.methodsS3_1.8.2         BPCells_0.1.0             progress_1.2.2            goftest_1.2-3             Biostrings_2.68.1         HDF5Array_1.28.1         
  [7] vctrs_0.6.3               spatstat.random_3.1-6     digest_0.6.33             png_0.1-8                 proxy_0.4-27              ggrepel_0.9.3            
 [13] deldir_1.0-9              magick_2.7.4              MASS_7.3-60               reshape2_1.4.4            httpuv_1.6.11             scico_1.5.0              
 [19] withr_2.5.1               ellipsis_0.3.2            survival_3.5-5            memoise_2.0.1             s2_1.1.4                  ggbeeswarm_0.7.2         
 [25] zoo_1.8-12                pbapply_1.7-2             R.oo_1.25.0               spData_2.3.0              prettyunits_1.2.0         rematch2_2.1.2           
 [31] KEGGREST_1.40.1           promises_1.2.1            httr_1.4.7                rstatix_0.7.2             globals_0.16.2            fitdistrplus_1.1-11      
 [37] rhdf5filters_1.12.1       rhdf5_2.44.0              rstudioapi_0.15.0         units_0.8-2               miniUI_0.1.1.1            generics_0.1.3           
 [43] curl_5.0.1                zlibbioc_1.46.0           ScaledMatrix_1.8.1        polyclip_1.10-6           GenomeInfoDbData_1.2.10   xtable_1.8-4             
 [49] doParallel_1.0.17         S4Arrays_1.0.6            BiocFileCache_2.8.0       hms_1.1.3                 irlba_2.3.5.1             colorspace_2.1-0         
 [55] filelock_1.0.2            ROCR_1.0-11               reticulate_1.32.0         spatstat.data_3.0-1       lmtest_0.9-40             spdep_1.2-8              
 [61] later_1.3.1               viridis_0.6.4             lattice_0.21-8            spatstat.geom_3.2-5       future.apply_1.11.0       scattermore_1.2          
 [67] XML_3.99-0.14             RcppAnnoy_0.0.21          class_7.3-22              pillar_1.9.0              nlme_3.1-162              iterators_1.0.14         
 [73] compiler_4.3.0            beachmat_2.16.0           RSpectra_0.16-1           stringi_1.7.12            tensor_1.5                plyr_1.8.9               
 [79] crayon_1.5.2              abind_1.4-5               locfit_1.5-9.8            graphlayouts_1.0.0        bit_4.0.5                 terra_1.7-29             
 [85] codetools_0.2-19          e1071_1.7-13              plotly_4.10.2             mime_0.12                 splines_4.3.0             Rcpp_1.0.11              
 [91] fastDummies_1.7.3         dbplyr_2.3.4              sparseMatrixStats_1.12.2  blob_1.2.4                utf8_1.2.3                listenv_0.9.0            
 [97] DelayedMatrixStats_1.22.6 ggsignif_0.6.4            statmod_1.5.0             tzdb_0.4.0                tweenr_2.0.2              pkgconfig_2.0.3          
[103] tools_4.3.0               cachem_1.0.8              RSQLite_2.3.1             viridisLite_0.4.2         DBI_1.1.3                 fastmap_1.1.1            
[109] scales_1.2.1              grid_4.3.0                ica_1.0-3                 broom_1.0.5               patchwork_1.1.3           BiocManager_1.30.22      
[115] dotCall64_1.0-2           carData_3.0-5             RANN_2.6.1                farver_2.1.1              tidygraph_1.2.3           wk_0.8.0                 
[121] cli_3.6.1                 leiden_0.4.3              lifecycle_1.0.3           uwot_0.1.16               backports_1.4.1           timechange_0.2.0         
[127] gtable_0.3.4              rjson_0.2.21              ggridges_0.5.4            progressr_0.14.0          parallel_4.3.0            limma_3.56.2             
[133] jsonlite_1.8.7            edgeR_3.42.4              RcppHNSW_0.5.0            bitops_1.0-7              bit64_4.0.5               assertthat_0.2.1         
[139] Rtsne_0.16                spatstat.utils_3.0-3      BiocNeighbors_1.18.0      zip_2.3.0                 metapod_1.8.0             dqrng_0.3.1              
[145] R.utils_2.12.2            lazyeval_0.2.2            shiny_1.7.5               htmltools_0.5.6           sctransform_0.4.0         rappdirs_0.3.3           
[151] glue_1.6.2                spam_2.9-1                XVector_0.40.0            RCurl_1.98-1.12           classInt_0.4-9            gridExtra_2.3            
[157] boot_1.3-28.1             igraph_1.4.3              R6_2.5.1                  cluster_2.1.4             Rhdf5lib_1.22.1           DelayedArray_0.26.7      
[163] tidyselect_1.2.0          vipor_0.4.5               ggforce_0.4.1             xml2_1.3.5                car_3.1-2                 AnnotationDbi_1.62.2     
[169] rsvd_1.0.5                munsell_0.5.0             KernSmooth_2.23-21        data.table_1.14.8         htmlwidgets_1.6.2         RColorBrewer_1.1-3       
[175] rlang_1.1.1               spatstat.sparse_3.0-2     spatstat.explore_3.2-3    remotes_2.4.2.1           ggnewscale_0.4.9          fansi_1.0.4              
[181] beeswarm_0.4.0           
```

**Comment by Fattigman** [comment]
Same problem here. Problem occurs only when giving the coordinate vector. No error when passing `x = NULL, y = NULL`. It looks like Seurat using a deprecated package under the hood?
```R
> sessionInfo()
R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Monterey 12.3.1

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Stockholm
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] patchwork_1.1.3    ggplot2_3.4.3      future_1.33.0      Seurat_4.4.0       dplyr_1.1.3        SeuratObject_4.1.4 sp_2.1-0          

loaded via a namespace (and not attached):
 [1] deldir_1.0-9           pbapply_1.7-2          gridExtra_2.3          rlang_1.1.1            magrittr_2.0.3         RcppAnnoy_0.0.21      
 [7] matrixStats_1.0.0      ggridges_0.5.4         compiler_4.3.1         spatstat.geom_3.2-5    png_0.1-8              vctrs_0.6.3           
[13] reshape2_1.4.4         stringr_1.5.0          pkgconfig_2.0.3        fastmap_1.1.1          ellipsis_0.3.2         labeling_0.4.3        
[19] utf8_1.2.3             promises_1.2.1         purrr_1.0.2            jsonlite_1.8.7         goftest_1.2-3          later_1.3.1           
[25] spatstat.utils_3.0-3   irlba_2.3.5.1          parallel_4.3.1         cluster_2.1.4          R6_2.5.1               ica_1.0-3             
[31] stringi_1.7.12         RColorBrewer_1.1-3     spatstat.data_3.0-1    reticulate_1.32.0      parallelly_1.36.0      lmtest_0.9-40         
[37] scattermore_1.2        Rcpp_1.0.11            tensor_1.5             future.apply_1.11.0    zoo_1.8-12             sctransform_0.4.0     
[43] httpuv_1.6.11          Matrix_1.6-1.1         splines_4.3.1          igraph_1.5.1           tidyselect_1.2.0       rstudioapi_0.15.0     
[49] abind_1.4-5            spatstat.random_3.1-6  codetools_0.2-19       miniUI_0.1.1.1         spatstat.explore_3.2-3 listenv_0.9.0         
[55] lattice_0.21-8         tibble_3.2.1           plyr_1.8.9             shiny_1.7.5            withr_2.5.1            ROCR_1.0-11           
[61] Rtsne_0.16             survival_3.5-5         polyclip_1.10-6        fitdistrplus_1.1-11    pillar_1.9.0           KernSmooth_2.23-21    
[67] plotly_4.10.2          generics_0.1.3         munsell_0.5.0          scales_1.2.1           globals_0.16.2         xtable_1.8-4          
[73] glue_1.6.2             lazyeval_0.2.2         tools_4.3.1            data.table_1.14.8      RANN_2.6.1             leiden_0.4.3          
[79] cowplot_1.1.1          grid_4.3.1             tidyr_1.3.0            colorspace_2.1-0       nlme_3.1-162           cli_3.6.1             
[85] spatstat.sparse_3.0-2  fansi_1.0.4            viridisLite_0.4.2      uwot_0.1.16            gtable_0.3.4           digest_0.6.33         
[91] progressr_0.14.0       ggrepel_0.9.3          farver_2.1.1           htmlwidgets_1.6.2      htmltools_0.5.6        lifecycle_1.0.3       
[97] httr_1.4.7             mime_0.12              MASS_7.3-60           

```

**Comment by dpcook** [comment]
Looks like update of `sp` to 2.1-0 earlier this month broke it.

Downgrading to 2.0-0 gets around the issue

```
devtools::install_version("sp", version = "2.0-0", repos = "http://cran.us.r-project.org")
```

```
obj[["CV1"]] <- Crop(obj[["fov"]], x = c(5485, 6077), y = c(583, 1180), 
+                         coords = "tissue")
Warning: No rgeos support in sp from October 2023;
see https://r-spatial.org/r/2023/05/15/evolution4.htmlWarning: 'over' is deprecated.
Use 'sf::st_intersects or sf::aggregate' instead.
See help("Deprecated")rgeos version: 0.6-4, (SVN revision 699)
 GEOS runtime version: 3.11.0-CAPI-1.17.0 
 Please note that rgeos will be retired during October 2023,
plan transition to sf or terra functions using GEOS at your earliest convenience.
See https://r-spatial.org/r/2023/05/15/evolution4.html for details.
 GEOS using OverlayNG
 Linking to sp version: 2.0-0 
 Polygon checking: TRUE 

Warning: Key ‘Xenium_’ taken, using ‘cv1_’ instead
```


```
> sessionInfo()
R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.4.1

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Toronto
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] Hmisc_5.1-1             InSituType_1.0.0       
 [3] matrixStats_1.0.0       Matrix_1.6-1.1         
 [5] Seurat_4.9.9.9059       SeuratObject_4.9.9.9091
 [7] sp_2.0-0                lubridate_1.9.3        
 [9] forcats_1.0.0           stringr_1.5.0          
[11] dplyr_1.1.3             purrr_1.0.2            
[13] readr_2.1.4             tidyr_1.3.0            
[15] tibble_3.2.1            ggplot2_3.4.3          
[17] tidyverse_2.0.0        

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.21            splines_4.3.1              
  [3] later_1.3.1                 bitops_1.0-7               
  [5] polyclip_1.10-6             rpart_4.1.19               
  [7] fastDummies_1.7.3           lifecycle_1.0.3            
  [9] globals_0.16.2              processx_3.8.2             
 [11] lattice_0.21-9              pals_1.8                   
 [13] MASS_7.3-60                 backports_1.4.1            
 [15] SnowballC_0.7.1             magrittr_2.0.3             
 [17] plotly_4.10.2               rmarkdown_2.25             
 [19] yaml_2.3.7                  remotes_2.4.2.1            
 [21] httpuv_1.6.11               sctransform_0.4.0          
 [23] askpass_1.2.0               spam_2.9-1                 
 [25] sessioninfo_1.2.2           pkgbuild_1.4.2             
 [27] spatstat.sparse_3.0-2       reticulate_1.32.0          
 [29] rgeos_0.6-4                 cowplot_1.1.1              
 [31] mapproj_1.2.11              pbapply_1.7-2              
 [33] RColorBrewer_1.1-3          zlibbioc_1.46.0            
 [35] maps_3.4.1                  abind_1.4-5                
 [37] pkgload_1.3.3               Rtsne_0.16                 
 [39] GenomicRanges_1.52.0        presto_1.0.0               
 [41] BiocGenerics_0.46.0         RCurl_1.98-1.12            
 [43] nnet_7.3-19                 GenomeInfoDbData_1.2.10    
 [45] IRanges_2.34.1              S4Vectors_0.38.2           
 [47] ggrepel_0.9.3               irlba_2.3.5.1              
 [49] listenv_0.9.0               spatstat.utils_3.0-3       
 [51] umap_0.2.10.0               goftest_1.2-3              
 [53] RSpectra_0.16-1             spatstat.random_3.1-6      
 [55] fitdistrplus_1.1-11         parallelly_1.36.0          
 [57] DelayedArray_0.26.7         leiden_0.4.3               
 [59] codetools_0.2-19            tidyselect_1.2.0           
 [61] farver_2.1.1                base64enc_0.1-3            
 [63] stats4_4.3.1                spatstat.explore_3.2-3     
 [65] jsonlite_1.8.7              Formula_1.2-5              
 [67] ellipsis_0.3.2              progressr_0.14.0           
 [69] ggridges_0.5.4              survival_3.5-7             
 [71] tools_4.3.1                 ica_1.0-3                  
 [73] Rcpp_1.0.11                 glue_1.6.2                 
 [75] gridExtra_2.3               xfun_0.40                  
 [77] MatrixGenerics_1.12.3       usethis_2.2.2              
 [79] GenomeInfoDb_1.36.3         withr_2.5.1                
 [81] fastmap_1.1.1               fansi_1.0.4                
 [83] openssl_2.1.1               callr_3.7.3                
 [85] digest_0.6.33               timechange_0.2.0           
 [87] R6_2.5.1                    mime_0.12                  
 [89] colorspace_2.1-0            scattermore_1.2            
 [91] tensor_1.5                  dichromat_2.0-0.1          
 [93] spatstat.data_3.0-1         utf8_1.2.3                 
 [95] generics_0.1.3              data.table_1.14.8          
 [97] S4Arrays_1.0.6              prettyunits_1.2.0          
 [99] httr_1.4.7                  htmlwidgets_1.6.2          
[101] uwot_0.1.16                 pkgconfig_2.0.3            
[103] gtable_0.3.4                lmtest_0.9-40              
[105] XVector_0.40.0              SingleCellExperiment_1.22.0
[107] htmltools_0.5.6             profvis_0.3.8              
[109] dotCall64_1.0-2             Biobase_2.60.0             
[111] scales_1.2.1                png_0.1-8                  
[113] knitr_1.44                  rstudioapi_0.15.0          
[115] tzdb_0.4.0                  reshape2_1.4.4             
[117] checkmate_2.2.0             nlme_3.1-163               
[119] cachem_1.0.8                zoo_1.8-12                 
[121] KernSmooth_2.23-22          parallel_4.3.1             
[123] miniUI_0.1.1.1              foreign_0.8-85             
[125] pillar_1.9.0                grid_4.3.1                 
[127] vctrs_0.6.3                 RANN_2.6.1                 
[129] lsa_0.73.3                  urlchecker_1.0.1           
[131] promises_1.2.1              xtable_1.8-4               
[133] cluster_2.1.4               htmlTable_2.4.1            
[135] evaluate_0.22               cli_3.6.1                  
[137] compiler_4.3.1              rlang_1.1.1                
[139] crayon_1.5.2                future.apply_1.11.0        
[141] labeling_0.4.3              mclust_6.0.0               
[143] ps_1.7.5                    plyr_1.8.9                 
[145] fs_1.6.3                    stringi_1.7.12             
[147] viridisLite_0.4.2           deldir_1.0-9               
[149] munsell_0.5.0               lazyeval_0.2.2             
[151] devtools_2.4.5              spatstat.geom_3.2-5        
[153] RcppHNSW_0.5.0              hms_1.1.3                  
[155] patchwork_1.1.3             future_1.33.0              
[157] shiny_1.7.5                 SummarizedExperiment_1.30.2
[159] ROCR_1.0-11                 igraph_1.5.1               
[161] memoise_2.0.1    
```

**Comment by Fattigman** [comment]
Thanks @dpcook, it works for me now!

**Comment by zskylarli** [comment]
This should be solved with Seurat 5.0.0 and SeuratObject 5.0.0. Please reinstall packages from CRAN.
